### PR TITLE
Requesting Signoff for Communication Subsystem

### PR DESCRIPTION
**Pull request by jacksonrwoodard**

### CODE REVIEW

File: Documentation/Signoffs/php-communication-signoff.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+![TransmitterCommunication](https://github.com/jacksonrwoodard/HouseHealthMonitoring/assets/142913669/cfb66cac-4edb-49bc-8843-ddd06d3072f7)
+
+![RecieverCommunication](https://github.com/jacksonrwoodard/HouseHealthMonitoring/assets/142913669/8ede150c-271b-4bfd-8071-dab157138e04)
+
+
+## Constraints
+| No. | Constraints                                                                                    | Origin                              |
+| --- | ---------------------------------------------------------------------------------------------- | ----------------------------------- |
+|  1  | Sensors shall not be placed in a way where they cannot wirelessly communicate with the system. | Project Team & Broader Implications |
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/php-communication-signoff.md]: Not specific.

File: Documentation/Signoffs/php-communication-signoff.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+|  2  | System shall not require an internet connection to work and communicate with the head unit and sensors | Home Owners, Insurance Agencies, & Team Supervisor |
+|  3  | Sensor communication shall send the sensor name, number, data type, and raw data to the head unit. | Project Team |
+|  4  | Sensor modules shall not send more than 64 bits of data when transmitting information to the head unit. | Project Team & Standards|
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/php-communication-signoff.md]: why?

**Review by JesseTNRoberts — COMMENTED** [review]
Why is there no constraint for rate?

**Review by JesseTNRoberts — COMMENTED** [review]
What sort of sensor needs a single connection for data to gpio?

**Comment by jacksonrwoodard** [comment]
> Why is there no constraint for rate?

We have added a constraint for rate and described why it is needed and how it will work.

File: Documentation/Signoffs/php-communication-signoff.md (hunk 1 above)
Comment (jacksonrwoodard) [reply Documentation/Signoffs/php-communication-signoff.md]: We updated this constraint to have specific ranges of communication.

File: Documentation/Signoffs/php-communication-signoff.md (hunk 2 above)
Comment (jacksonrwoodard) [reply Documentation/Signoffs/php-communication-signoff.md]: > why?

We explained why the sensors shall not send more than 64 bits of data underneath the constraints section on number 6.

**Comment by jacksonrwoodard** [comment]
> What sort of sensor needs a single connection for data to gpio?

We corrected the third-party buildable schematic. The sensors that we are using do not require more than two pins to send data. Will be further explained in each sensor module.

File: Documentation/Signoffs/php-communication-signoff.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+![TransmitterCommunication](https://github.com/jacksonrwoodard/HouseHealthMonitoring/assets/142913669/cfb66cac-4edb-49bc-8843-ddd06d3072f7)
+
+![RecieverCommunication](https://github.com/jacksonrwoodard/HouseHealthMonitoring/assets/142913669/8ede150c-271b-4bfd-8071-dab157138e04)
+
+
+## Constraints
+| No. | Constraints                                                                                    | Origin                              |
+| --- | ---------------------------------------------------------------------------------------------- | ----------------------------------- |
+|  1  | System shall not require an internet connection to work and communicate with the head unit and sensors | Home Owners, Insurance Agencies, & Team Supervisor |
+|  2  | At least one sensor shall be placed within 200ft of the head unit. | Project Team & Broader Implications & Standards |
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/php-communication-signoff.md]: How is this a broader/standard constraint?

File: Documentation/Signoffs/php-communication-signoff.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+|  3  | Sensors shall be within 32ft from each other for every sensor outside of 200ft of the head unit. | Project Team & Broader Implications & Standards |
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/php-communication-signoff.md]: Same question.

File: Documentation/Signoffs/php-communication-signoff.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+<sup>1</sup> In order for the system to not require an internet connection, the system will be designed to work on a Personal Area Network (PAN). In order to achieve this, the Zigbee communication protocol will be used.
+
+<sup>2</sup> In order for the system to work properly all of the sensors have to be able to communicate with the head unit. At least one sensor has to be within 200ft of the head unit because the XB24's maximum range allows for 200ft of indoor communication. Indoor means through the interior walls of a house.
+
+<sup>3</sup> The sensors have to be placed within 32ft of each other for the minimum transmission and reception range of the Zigbee-compliant devices.
+
+<sup>4</sup> The head unit needs to know the sensor information to be able to organize the data of each sensor. Knowing this allows the information to be more easily displayed to the user.
+
+<sup>5</sup> This is the maximum amount of data that Zigbee can send per data request from the head unit. This is also necessary to preserve storage and power consumption. It will also prevent overflow and loss of information.
```
Comment (JesseTNRoberts) [inline Documentation/Signoffs/php-communication-signoff.md]: What does this mean? Why is it a constraint?

**Review by JesseTNRoberts — COMMENTED** [review]
You haven't addressed how much data you need to send or how fast in the constraints.

File: Documentation/Signoffs/php-communication-signoff.md (hunk 3 above)
Comment (jacksonrwoodard) [reply Documentation/Signoffs/php-communication-signoff.md]: We changed the constraint and meant to take out broader/standard. It is fixed now.

File: Documentation/Signoffs/php-communication-signoff.md (hunk 4 above)
Comment (jacksonrwoodard) [reply Documentation/Signoffs/php-communication-signoff.md]: We changed the constraint and meant to take out broader/standard. It is fixed now.

**Comment by jacksonrwoodard** [comment]
> You haven't addressed how much data you need to send or how fast in the constraints.

We created a new constraint and explained why it is needed and proved that it is possible.

File: Documentation/Signoffs/php-communication-signoff.md (hunk 5 above)
Comment (jacksonrwoodard) [reply Documentation/Signoffs/php-communication-signoff.md]: We reworded this constraint to make it related to amount of data sent and the speed.

**Review by JesseTNRoberts — COMMENTED** [review]
Based on what is once in 24 hours sufficient for this sensing application?

Why does the base use a different device for comm than the sensors?

**Review by JesseTNRoberts — COMMENTED** [review]
How do you know that the esp will be sufficient to intake all the different comms of other devices? 

It doesn't look like the xbee chip has tx and rx

**Comment by docrobbins4** [comment]
> Based on what is once in 24 hours sufficient for this sensing application?
> 
> Why does the base use a different device for comm than the sensors?

Both of these are addressed in the analysis section. An explanation of why 24 hours is sufficient for each data transmission is given. Also, the team addressed other potential cases when data shall be transmitted to the head unit. An explanation of why a different device is used for the base is also provided.

**Comment by docrobbins4** [comment]
> How do you know that the esp will be sufficient to intake all the different comms of other devices?
> 
> It doesn't look like the xbee chip has tx and rx

The team addressed how the ESPs will be sufficient to intake, store, and transmit communication between other devices in the analysis section. An example of proof is also provided. The explanation of the TX and RX pins on the XB24 is provided beneath the schematic.

**Review by JesseTNRoberts — COMMENTED** [review]
Why is the ESP 32 being used? Consider a cheaper microcontroller. I have 3 IOT 33 that can be used.

I am happy to approve this, however, please plan to use the 3 IOT 33 microcontrollers we have rather than purchasing the ESP32.

**Review by JesseTNRoberts — APPROVED** [review]

**Pull request merged**